File: leetcode/medium/1927. Sum Game/main.py
class Solution:
    def sumGame(self, num: str) -> bool:
        half = len(num) // 2

        sum_diff, q_diff = 0, 0

        for i, v in enumerate(num):
            if i < half:
                if v == '?':
                    q_diff += 1
                else:
                    sum_diff += int(v)
            else:
                if v == '?':
                    q_diff -= 1
                else:
                    sum_diff -= int(v)

        return (sum_diff * 2 + q_diff * 9) != 0

File: leetcode/medium/1927. Sum Game/main.go
package main

func sumGame(num string) bool {
	var (
		sumDiff int
		qDiff   int
	)

	for i, v := range num {
		if i < len(num)/2 {
			if v == '?' {
				qDiff++
			} else {
				sumDiff += int(v) - 48
			}
		} else {
			if v == '?' {
				qDiff--
			} else {
				sumDiff -= int(v) - 48
			}
		}
	}

	return (sumDiff*2 + qDiff*9) != 0
}
